check.go

```diff
@@ -899,11 +930,20 @@ func (c *checkDirectory) add(path string, typeflag byte, uid, gid int, size int6
 	if components[len(components)-1] == "" {
 		components = components[:len(components)-1]
 	}
-	if typeflag == tar.TypeLink || typeflag == tar.TypeRegA {
+	if components[0] == "." {
+		components = components[1:]
+	}
+	if typeflag == tar.TypeRegA {
 		typeflag = tar.TypeReg
```

Comment:
`archive/tar.Reader.next` can possibly turn `TypeRegA` into `TypeDir` based on trailing slash in `hdr.Name`. Is that applicable here?

Reply:
These either come from `Next()`, which won't give us a header with `TypeRegA` in it, or from `tar.FileInfoHeader()`, which either sets a different type flag or returns an error, so I guess not.